libpod/container_api.go

```diff
@@ -218,48 +220,28 @@ func (c *Container) Kill(signal uint) error {
 
 // Exec starts a new process inside the container
 // TODO investigate allowing exec without attaching
-func (c *Container) Exec(tty, privileged bool, env, cmd []string, user, workDir string, streams *AttachStreams, preserveFDs int) error {
+func (c *Container) Exec(tty, privileged bool, env, cmd []string, user, workDir string, streams *AttachStreams, preserveFDs int, resize chan remotecommand.TerminalSize, detachKeys string) (int, error) {
 	var capList []string
-
-	locked := false
 	if !c.batched {
-		locked = true
```

Comment:
We need this if you're going to keep that defer around - if not, we can double unlock and crash

Reply:
By defer I mean the `defer lock.Unlock()`

Reply:
I don't think there's a way to double lock. The only way the function returns between the manual calls of c.lock.Unlock() is a runtime panic, which wouldn't call deferred functions anyway, right? The locked value was to make sure we didn't return when unlocked before when the code was more complicated here, I think